sdk/src/main/scala/com/horizen/block/MainchainBlockReference.scala

```diff
@@ -31,19 +31,21 @@ import scala.collection.mutable
 
 @JsonView(Array(classOf[Views.Default]))
 @JsonIgnoreProperties(Array("hash", "hashHex"))
-class MainchainBlockReference(
-                    val header: MainchainHeader,
-                    val sidechainRelatedAggregatedTransaction: Option[MC2SCAggregatedTransaction],
-                    @JsonProperty("merkleRoots")
-                    @JsonSerialize(using = classOf[JsonMerkleRootsSerializer])
-                    val sidechainsMerkleRootsMap: Option[mutable.Map[ByteArrayWrapper, Array[Byte]]]
+case class MainchainBlockReference(
+                    header: MainchainHeader,
+                    data: MainchainBlockReferenceData
```

Comment:
At the moment we verify that `MainchainHeaders` included in the `SidechainBlocks` are semantically valid, ordered in a chain and have valid PoW.

Because of `MainchainBlockReference` separation, need to add `HistoryBlockValidator` that will validate that `MainchainBlockReferenceData` included into `SidechainBlock` are consistent to `MainchainHeaders` and together with them lead to valid `MainchainBlockReference`.